packages/shared/src/components/modals/firefoxPrivacy/FirefoxPrivacyModal.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement } from 'react';
+import { Modal, ModalProps } from '../common/Modal';
+import { ModalSize } from '../common/types';
+import { ButtonVariant } from '../../buttons/common';
+import { Button } from '../../buttons/Button';
+
+export interface FirefoxPrivacyModalProps extends ModalProps {
+  onAccept: () => void;
+  onDecline: () => void;
+}
+
+function FirefoxPrivacyModal({
+  onAccept,
+  onDecline,
+  onRequestClose,
+  ...props
+}: FirefoxPrivacyModalProps): ReactElement {
+  const actionAndClose = (action: () => void, e: React.MouseEvent) => {
+    action();
+    onRequestClose(e);
+  };
+
+  return (
+    <Modal {...props} shouldCloseOnOverlayClick={false} size={ModalSize.Medium}>
+      <Modal.Body>
+        <h1 className="text-center font-bold typo-title3">
+          Your privacy stays yours
+        </h1>
+        <div className="mb-6 mt-4 text-center text-text-secondary typo-callout">
+          <p>
+            We collect user data to provide a personalized experience on
+            daily.dev. Instead of just collecting data right away, we would like
+            to ask for your approval. We promise to never misuse it.
```

Comment:
Maybe add link to our privacy policy?

Reply:
I chatted to Nimrod before he said it was fine not doing so but raised it again 🙏

Reply:
I predict FF might react that there's no link to the policy.

Reply:
Added for now :)